### PR TITLE
Samiksha survey apis

### PR DESCRIPTION
**Pull request by praveenKDass**

### CODE REVIEW

File: controllers/v1/surveySubmissionsController.js [hunk 1]
```diff
@@ -409,4 +409,258 @@ module.exports = class SurveySubmissions extends Abstract {
...
+
+   /**
+   * update survey submissions.
+   * @method
+   * @name update
+   * @param {Object} req -request data.
+   * @returns {JSON} - survey submissions creation.
+   */
+
+   async update(req) {
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/surveySubmissionsController.js:596]: @praveenKDass All the logic is in controller . please avoid this. we should move them to helper, db query files accordingly. Please do this for all changes in this PR

File: module/programs/helper.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ module.exports = class ProgramsHelper {
    * @returns {Object} - Programs list.
    */
 
-  static list(pageNo = '', pageSize = '', searchText, filter = {}, projection) {
+  static list(filter = {}, projection,pageNo = '', pageSize = '', searchText ) {
```
Comment (VISHNUDAS-tunerlabs) [inline module/programs/helper.js:37]: @praveenKDass change in order of param won't affect places where this function being called?

File: module/surveys/validator/v1.js [hunk 3]
```diff
@@ -27,10 +27,6 @@ module.exports = (req) => {
     getDetailsByLink: function () {
       req.checkParams('_id').exists().withMessage('required link');
     },
-
-    details: function () {
```
Comment (VISHNUDAS-tunerlabs) [inline module/surveys/validator/v1.js:31]: @praveenKDass we don't need this?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

File: module/programs/helper.js (hunk 2 above)
Comment (praveenKDass) [reply module/programs/helper.js:37]: No ,verified and modified with other places where its used , The thing is in other helper files where ever this function called its passing as above params @VISHNUDAS-tunerlabs

File: module/surveys/validator/v1.js (hunk 3 above)
Comment (praveenKDass) [reply module/surveys/validator/v1.js:31]: Yes we do't need this as details can be called by passing surveyId or solution id or through link @VISHNUDAS-tunerlabs

File: controllers/v1/surveySubmissionsController.js (hunk 1 above)
Comment (praveenKDass) [reply controllers/v1/surveySubmissionsController.js:353]: Modified as per the requirements @VISHNUDAS-tunerlabs

**Pull request merged**